### PR TITLE
Add bastion architecture diagram in docs/

### PR DESCRIPTION
**Pull request by ya5ut**
This PR adds a diagram for bastion's system architecture.

The goals are:
- new developers can more easily figure out how bastion manages jobs and logs
- developers can discuss how to organize bastion-related modules

See preview on [the bastion.md file's "View File" page](https://github.com/apple/axlearn/blob/1de793e44b486020f0e74ed42a3ce779adfe261c/docs/bastion.md).

### CODE REVIEW

File: docs/bastion.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+## Control Flow of Job Submission
+```mermaid
+%% elk seems to be more maintained, see: https://github.com/mermaid-js/mermaid/issues/1969
+%% N.B. elk doesn't stop rendering invisible edge operator, i.e. ~~~
+%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%
+
+flowchart TB
+
+    subgraph UserMachine ["User's dev machine (e.g. MacBook Pro)"]
+        subgraph AXLearnGithubRepository["AXLearn Github Repository"]
```
Comment (markblee) [inline docs/bastion.md:12]: Actually, submit need not start from github repo. We should be able to launch from a pip install in a different CWD.

File: docs/bastion.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+            workerProcess_2 --> accelerator_2
+        end
+
+        bastionLogStore[("Log Store \n (e.g. Google Storage)")]:::storeCSS
+        WorkerVM_2 --"sync logs"--> bastionLogStore
+
+    end
+
+    bastionLogStore--"download logs for debug"-->UserMachine
```
Comment (markblee) [inline docs/bastion.md:65]:
```suggestion
    bastionLogStore--"Download logs for debug"-->UserMachine
```

File: docs/bastion.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        subgraph WorkerVM_2 ["Worker VM 2"]
+            workerProcess_2["..."]
+            accelerator_2[/"..."\]:::chipCSS
+            bastionWorkerAXLearnPackage_2(["..."]):::fileCSS
+
+            workerProcess_2 --> accelerator_2
+        end
+
+        bastionLogStore[("Log Store \n (e.g. Google Storage)")]:::storeCSS
+        WorkerVM_2 --"sync logs"--> bastionLogStore
```
Comment (markblee) [inline docs/bastion.md:61]:
```suggestion
        WorkerVM_2 --"Sync logs"--> bastionLogStore
```
Should WorkerVM_1 also have the "sync logs" branch?

File: docs/bastion.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    subgraph PublicCloud ["Public Cloud (e.g. Google Cloud Platform)"]
+
+        subgraph BastionVM_shared ["Bastion VM (e.g. 'shared-bastion')"]
+            bastionScheduler_1["Bastion \n Scheduler"]
+            bastionVmAXLearnPackage(["axlearn package \n (running shared docker image)"]):::fileCSS
+            bastionJob_1["Bastion job 1 \n name: notebook-tpu-alice-a59ce1"]
+            bastionJob_2["Bastion job 2 \n name: notebook-tpu-bob-b3b5f1"]
+
+            bastionScheduler_1 --"spawn/kill"--> bastionJob_1 & bastionJob_2
```
Comment (markblee) [inline docs/bastion.md:34]:
```suggestion
            bastionScheduler_1 --"Spawn/kill"--> bastionJob_1 & bastionJob_2
```

File: docs/bastion.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    localAXLearnPackage =="
+        Submit a bastion job
+        (serialized as a job spec)"==> bastionPrimaryStore
+
+    subgraph PublicCloud ["Public Cloud (e.g. Google Cloud Platform)"]
+
+        subgraph BastionVM_shared ["Bastion VM (e.g. 'shared-bastion')"]
+            bastionScheduler_1["Bastion \n Scheduler"]
+            bastionVmAXLearnPackage(["axlearn package \n (running shared docker image)"]):::fileCSS
```
Comment (markblee) [inline docs/bastion.md:30]:
```suggestion
            bastionVmAXLearnPackage(["axlearn package \n (running on shared docker image)"]):::fileCSS
```

**Review by markblee — APPROVED** [review]
Thanks! Mostly nits.

File: docs/bastion.md (hunk 3 above)
Comment (ya5ut) [reply docs/bastion.md:61]: It can, I just omitted it for simplicity

**Review by markblee — APPROVED** [review]

**Pull request merged**